src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -4257,6 +4260,18 @@ public void testCreateService() throws Exception {
     final ServiceCreateResponse response = sut.createService(spec);
     assertThat(response.id(), is(notNullValue()));
   }
+  
+  @Test
+  public void testCreateSecret() throws Exception {
+    requireDockerApiVersionAtLeast("1.25", "swarm support");
+    
+    final String secretData = Base64.encodeAsString("testdata".getBytes());
```

Comment:
Could we also test what happens when the data isn't Base64-url-safe-encoded, with labels, collision with an existing secret, etc?